rfc/2020/conformance-validation/mayankshah1607/rfc.md

```diff
@@ -0,0 +1,477 @@
+
+**2. A sonobuoy plugin file (YAML)**
+
+The main function of Sonobuoy is running plugins; each plugin may run tests or gather data in the cluster. A plugin is defined using a YAML file that holds information such as 
+
+- Name of plugin
+- Format of result
+- Type of a plugin - Job or DaemonSet
+- Command to execute
+- args/flags for the the command
+- The Docker image to use
+- volumeMounts - for storing results
+
+We shall provide a plugin file, _l5d_conformance.yaml_ for running our test framework as a Sonobuoy plugin. It may look something like :
+```yaml
+sonobuoy-config:
+  driver: Job
+  plugin-name: l5d-conformance
+  result-format: raw
+spec:
+  args:
+  - --enableTest="TestAutoProxyInjection/*" \
+  - --failFast \
+  - --ha \
+  - --addon-config=addon-config.yaml
+  # flags should be added here. More information regarding these is described below
+  command:
+  - ./run.sh
+  image: gcr.io/linkerd.io/conformance:v1.0.0
+  name: plugin
+  resources: {}
+  volumeMounts:
+  - mountPath: /tmp/results
+    name: results
+``` 
+
+In order to run the Sonobuoy plugin, users/testers shall be required to carry out the following steps:
+
+1. Edit the above mentioned plugin file as required. This involves modifying the list of args.
+
+2. Run the plugin
+```bash
+$ sonobuoy run --plugin l5d-conformance.yaml
+```
+3. Collect output
+```
+$ outfile=$(sonobuoy retrieve) && \
+mkdir results && tar -xf $outfile -C results
+```
+---
+
+### The role of `run.sh`
+The `run.sh` file performs the following functionalities :
+
+#### 1. Build and issue a `go test` command
+The executable `run.sh` file shall accept flags and build a `go test` cmd to make the tests configurable as well as set `linkerd install` options.
+
+Usage : 
+```bash
+./run.sh [flags]
+```
+Flags :
+
+**(Specific to test configuration)**
+
+``` 
+--wait String
+    duration for which the tests must wait for pods to come to a running state
+
+--enableTests String
+    Comma separated list of test names that must be executed. This is later converted into regexp and passed as a flag value for `-run` for `go test` command.
+
+--skipTests String
+    Comma separated list of test names that must be skipped. The value of this is later used in the go code with the t.Skip() method.
+
+--failFast
+    Do not start new tests after the first test failure.
+
+--parallel int
+    Number of tests to run in parallel
+
+--buildTags String
+    Comma separated list of tags that must be included in the test
+
+--o String
+    Filepath to store the result. Ignored in case of Sonobuoy 
+
+--skipInstall
+    Skips the `linkerd install` process. This is useful for testers who already have linkerd installed prior to running the tests
+
+--addon-config String
+    Filepath to the addon config file. This flag must be used only in "default mode"
+
+--addon-config-sonobuoy String
```

Comment:
I don't think this is necessary. See comment above on adding a `COPY` to the Dockerfile.